### PR TITLE
chore(deps): dump feel-engine from 1.12.4 to 1.13.1

### PR DESCRIPTION
**Pull request by saig0**
## Description

* update `feel-engine` to [1.13.1](https://github.com/camunda/feel-scala/releases/tag/1.13.1) (see also the changelog of [1.13.0](https://github.com/camunda/feel-scala/releases/tag/1.13.0))
* replace the custom function `appendTo()`
  * currently, the output variable mapping uses the custom function `appendTo()` to merge nested contexts
  * replace the custom function by the new builtin function [put all()](https://github.com/camunda/feel-scala/pull/239)

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by korthout** [comment]
please don't forget to document in the release notes

File: engine/src/main/java/io/zeebe/engine/processing/deployment/model/validation/ExpectedValidationResult.java [hunk 1]
```diff
@@ -56,7 +56,7 @@ public boolean matches(final ValidationResult result) {
       match = false;
     }
 
-    if (!expectedMessage.equals(result.getMessage())) {
+    if (!result.getMessage().startsWith(expectedMessage)) {
```
Comment (korthout) [inline engine/src/main/java/io/zeebe/engine/processing/deployment/model/validation/ExpectedValidationResult.java:59]: why is this okay?

File: expression-language/src/main/scala/io/zeebe/el/impl/feel/FeelFunctionProvider.scala [hunk 2]
```diff
@@ -7,41 +7,19 @@
...
-import org.camunda.feel.context.{Context, FunctionProvider}
+import org.camunda.feel.context.FunctionProvider
 import org.camunda.feel.syntaxtree._
 
 class FeelFunctionProvider extends FunctionProvider {
 
   override lazy val functionNames: Iterable[String] = functions.keys
 
   private val functions: Map[String, List[ValFunction]] = Map(
-    "appendTo" -> List(appendFunction),
```
Comment (korthout) [inline expression-language/src/main/scala/io/zeebe/el/impl/feel/FeelFunctionProvider.scala:19]: yay 🎉

**Review by korthout — APPROVED** [review]
LGTM :octocat: but to be sure please have a look at the question @saig0 (context: it feels as if this validation change is to support some simplified tests, but shouldn't be a production code change)

File: engine/src/main/java/io/zeebe/engine/processing/deployment/model/validation/ExpectedValidationResult.java (hunk 1 above)
Comment (saig0) [reply engine/src/main/java/io/zeebe/engine/processing/deployment/model/validation/ExpectedValidationResult.java:59]: Good catch :eyes: The class should be used in tests only. I'll move it. 

I changed the verification to simplify the tests (i.e. don't verify the whole message) and to make it more stable for updates (i.e. if the error message changes in the FEEL engine).

**Pull request merged**